api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -122,27 +122,30 @@
    *
    * @param conflictDetectionFilter an expression on rows in the table
    * @return this for method chaining
+   * @deprecated this will be removed in 0.14.0;
+   *             use {@link #validateNoConflictingOperations(Expression)} instead
    */
-  OverwriteFiles validateNoConflictingAppends(Expression conflictDetectionFilter);
+  @Deprecated
+  default OverwriteFiles validateNoConflictingAppends(Expression conflictDetectionFilter) {
+    return validateNoConflictingOperations(conflictDetectionFilter);
+  }
 
   /**
-   * Enables validation that files added concurrently do not conflict with this commit's operation.
+   * Enables validation that no concurrent operation conflicts with this commit.
    * <p>
    * This method should be called when the table is queried to determine which files to delete/append.
-   * If a concurrent operation commits a new file after the data was read and that file might
-   * contain rows matching the specified conflict detection filter, the overwrite operation
-   * will detect this during retries and fail.
+   * If a concurrent operation commits a new data or delete file after the table was queried and
+   * that file might contain either new records or relevant deletes matching the specified conflict
+   * detection filter, the overwrite operation will detect this during retries and fail.
    * <p>
    * Calling this method with a correct conflict detection filter is required to maintain
    * serializable isolation for eager update/delete operations. Otherwise, the isolation level
    * will be snapshot isolation.
+   * <p>
+   * Validation applies to operations happened since the snapshot passed to {@link #validateFromSnapshot(long)}.
```

Comment:
Nit: This is missing a "that" before "happened". But then it sounds odd to me, so I'd recommend "that happened after" instead of "happened since".

Reply:
Fixed.